code/datums/looping_sounds/_looping_sound.dm

```diff
@@ -0,0 +1,94 @@
+/*
+	output_atoms	(list of atoms)			The destination(s) for the sounds
+
+	mid_sounds		(list or soundfile)		Since this can be either a list or a single soundfile you can have random sounds. May contain further lists but must contain a soundfile at the end.
+	mid_length		(num)					The length to wait between playing mid_sounds
+
+	start_sound		(soundfile)				Played before starting the mid_sounds loop
+	start_length	(num)					How long to wait before starting the main loop after playing start_sound
+
+	end_sound		(soundfile)				The sound played after the main loop has concluded
+
+	chance			(num)					Chance per loop to play a mid_sound
+	volume			(num)					Sound output volume
+	max_loops		(num)					The max amount of loops to run for.
+	direct			(bool)					If true plays directly to provided atoms instead of from them
+*/
+
+/datum/looping_sound
```

Comment:
As discussed, I think `composite_sound` or `dynamic_sound` would be a less confusing name. I also think it would be good for this to be split into a separate PR for review as it's pretty distinct from the body of the cooking system and could be useful generally even if the cooking system ends up being a poor fit for whatever reason.